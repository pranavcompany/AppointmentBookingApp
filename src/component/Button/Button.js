import React from 'react';
import PropTypes from 'prop-types';
import Icon from 'react-native-vector-icons/FontAwesome';
import {Label, ButtonWrapper} from './Button.style';

const Button = ({
  onPress,
  icon,
  iconName,
  iconColor,
  label,
  testID,
  labelColor,
  backgroundColor,
  margin,
}) => {
  return (
    <ButtonWrapper
      testID={testID}
      onPress={onPress}
      backgroundColor={backgroundColor}
      margin={margin}>
      {icon ? (
        <Icon name={iconName} color={iconColor} size={30} />
      ) : (
        <Label color={labelColor}>{label}</Label>
      )}
    </ButtonWrapper>
  );
};

Button.prototype = {
  onPress: PropTypes.func,
  icon: PropTypes.bool,
  iconName: PropTypes.string,
  iconColor: PropTypes.string,
  testID: PropTypes.string,
  labelColor: PropTypes.string,
  backgroundColor: PropTypes.string,
  margin: PropTypes.number,
};

Button.defaultProps = {
  iconName: 'angle-left',
  iconColor: '#fff',
  testID: 'button',
  labelColor: '#fff',
  backgroundColor: 'transparent',
};
export default Button;
