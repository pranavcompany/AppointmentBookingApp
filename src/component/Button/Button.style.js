import styled from 'styled-components/native';
import {Text} from 'react-native';

export const Label = styled(Text)`
  font-size: ${props => props.fontSize || 16}px;
  text-align: center;
  margin: ${props => props.margin || 15}px;
  color: ${props => props.color || '#66b3ff'};
`;

export const ButtonWrapper = styled.TouchableOpacity`
  border-radius: 6px;
  align-items: center;
  justify-content: center;
  ${({backgroundColor}) =>
    backgroundColor && ` background-color:${backgroundColor}`}
  ${({marginTop}) => marginTop && `margin-top:${marginTop}px`};
  ${({margin}) => margin && `margin:${margin}px`};
  ${({marginBottom}) => marginBottom && `margin-bottom:${marginBottom}px`};
`;
