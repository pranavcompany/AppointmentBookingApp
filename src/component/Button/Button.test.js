import React from 'react';
import {render} from '@testing-library/react-native';
import Button from './Button';

describe('Button ', () => {
  it('Should render button icon', () => {
    const tree = render(
      <Button
        testID="back-button"
        icon
        iconColor="#000"
        iconName="angle-left"
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Should render button label', () => {
    const tree = render(
      <Button
        backgroundColor="#33adff"
        testID="button-2"
        margin={10}
        label="demo"
      />,
    );
    expect(tree).toMatchSnapshot();
  });
});
