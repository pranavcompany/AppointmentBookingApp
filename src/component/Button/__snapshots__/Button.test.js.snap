// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button  Should render button icon 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderRadius": 6,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID="back-button"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 30,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï„„
  </Text>
</View>
`;

exports[`Button  Should render button label 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#33adff",
      "borderRadius": 6,
      "justifyContent": "center",
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "opacity": 1,
    }
  }
  testID="button-2"
>
  <Text
    color="#fff"
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 16,
          "marginBottom": 15,
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 15,
          "textAlign": "center",
        },
      ]
    }
  >
    demo
  </Text>
</View>
`;
